node/cn/peer_set_test.go

```diff
@@ -524,6 +524,127 @@ func TestPeerSet_SampleResendPeersByType_EN(t *testing.T) {
 	}
 }
 
+func TestPeerSet_SampleResendPeersByType_Default(t *testing.T) {
+	peerSet := newPeerSet()
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	cnPeer1 := NewMockPeer(mockCtrl)
+	cnPeer2 := NewMockPeer(mockCtrl)
+	pnPeer1 := NewMockPeer(mockCtrl)
+	pnPeer2 := NewMockPeer(mockCtrl)
+	enPeer1 := NewMockPeer(mockCtrl)
+	enPeer2 := NewMockPeer(mockCtrl)
+
+	setMockPeersConnType(cnPeer1, pnPeer1, enPeer1)
+	setMockPeersConnType(cnPeer2, pnPeer2, enPeer2)
+	setMockPeers([]*MockPeer{cnPeer1, pnPeer1, enPeer1, cnPeer2, pnPeer2, enPeer2})
+
+	assert.NoError(t, peerSet.Register(cnPeer1))
+	assert.NoError(t, peerSet.Register(pnPeer1))
+	assert.NoError(t, peerSet.Register(enPeer1))
+
+	assert.NoError(t, peerSet.Register(cnPeer2))
+	assert.NoError(t, peerSet.Register(pnPeer2))
+	assert.NoError(t, peerSet.Register(enPeer2))
+
+	assert.Nil(t, peerSet.SampleResendPeersByType(node.UNKNOWNNODE))
+	assert.Nil(t, peerSet.SampleResendPeersByType(node.BOOTNODE))
+	assert.Nil(t, peerSet.SampleResendPeersByType(node.CONSENSUSNODE))
+}
+
+func TestPeerSet_PeersWithoutBlockExceptCN(t *testing.T) {
+	peerSet := newPeerSet()
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	cnPeer1 := NewMockPeer(mockCtrl)
+	cnPeer2 := NewMockPeer(mockCtrl)
+	pnPeer1 := NewMockPeer(mockCtrl)
+	pnPeer2 := NewMockPeer(mockCtrl)
+	enPeer1 := NewMockPeer(mockCtrl)
+	enPeer2 := NewMockPeer(mockCtrl)
+
+	setMockPeersConnType(cnPeer1, pnPeer1, enPeer1)
+	setMockPeersConnType(cnPeer2, pnPeer2, enPeer2)
+	setMockPeers([]*MockPeer{cnPeer1, pnPeer1, enPeer1, cnPeer2, pnPeer2, enPeer2})
+
+	assert.NoError(t, peerSet.Register(cnPeer1))
+	assert.NoError(t, peerSet.Register(pnPeer1))
+	assert.NoError(t, peerSet.Register(enPeer1))
+
+	assert.NoError(t, peerSet.Register(cnPeer2))
+	assert.NoError(t, peerSet.Register(pnPeer2))
+	assert.NoError(t, peerSet.Register(enPeer2))
+
+	block := newBlock(blockNum1)
+	pnPeer1.EXPECT().KnowsBlock(block.Hash()).Return(true).AnyTimes()
+	pnPeer2.EXPECT().KnowsBlock(block.Hash()).Return(false).AnyTimes()
+
+	enPeer1.EXPECT().KnowsBlock(block.Hash()).Return(false).AnyTimes()
+	enPeer2.EXPECT().KnowsBlock(block.Hash()).Return(true).AnyTimes()
+
+	result := peerSet.PeersWithoutBlockExceptCN(block.Hash())
+	assert.Equal(t, 2, len(result))
+	assert.False(t, containsPeer(pnPeer1, result))
```

Comment:
How about checking `cnPeer1,2` is not contained?

Reply:
I omitted checking cnPeer1 and cnPeer2 since line 590 and 591 checks there are two peers (pnPeer2, enPeer1) and the length of the result is 2 by line 588. Do we need additional explicit check for cnPeer1 and cnPeer2?

Reply:
As you said, present code has same effect logically. But `PeersWithoutBlockExceptCN`'s feature is excepting CN`. So I think explicitly checking it is better.

Reply:
@KimKyungup Updated!